Add Access List documentation. (#31781)
gravitational/teleport: docs/pages/access-controls/access-lists/guide.mdx

---
title: Getting Started with Access Lists
description: Learn how to use Access Lists to manage and audit long lived access to Teleport managed resources.
---

This guide will help you:
- Create an access list
- Assign a member to it
- Verify permissions granted through the list membership

## Prerequisites

(!docs/pages/includes/commercial-prereqs-tabs.mdx!)

- (!docs/pages/includes/tctl.mdx!)
- A running Teleport cluster with some built in resources. Following the

>   [Web Application Access Guide](../../application-access/guides/connecting-apps.mdx) will create a base with a

Reviewer: The guide just tells you that you need some resources and kind of assumes that you don't have access to the resource before being added to the access list and that you do have access after being added. I wonder if there's a way to make this more concrete.

Author: I'm sure there is. What do you think about making the `Connecting Apps` guide a little more fully featured in this doc? We effectively need to instruct the user, "create a few protected resources" somehow.

Author: I've added in a step to add the dumper app to the cluster. Let me know what you think.

---

> title: A Simple Access List Example

Reviewer: Should this be called getting started instead?

description: Learn how to use Access Lists to manage and audit long lived access to Teleport managed resources.
---

> <Admonition type="warning">
> Access Lists are currently in Preview mode.
> </Admonition>

Reviewer 2: ```suggestion
```

> This guide will help you set up a basic Access List.

Reviewer 2: ```suggestion
This guide will help you:
- Create an access list
- Assign a member to it
- Verify permissions granted through the list membership
```

## Prerequisites

(!docs/pages/includes/commercial-prereqs-tabs.mdx!)

- (!docs/pages/includes/tctl.mdx!)
- A running Teleport cluster with some built in resources. Following the
  [Web Application Access Guide](../../application-access/guides/connecting-apps.mdx) will create a base with a
  simple application that we can use to demonstrate access lists.
- An administrative user with the preset `editor` role, which will have permissions to create Access Lists.

## Step 1/3. Create a test user

First, we need to create a simple test user that has only the `requester` role, which has no default access
to anything within a cluster. This user will only be used for the purposes of this guide, so you may use
another user if you so choose.

```code

> $ tctl users add test-user --roles=requester --logins=root,ubuntu,ec2-user

Reviewer 2: Can we replace this with the instructions of inviting the user using web UI?

```

> Then, follow the link provided and create credentials for this user. Try logging in to verify that no resources

Reviewer 2: Will the login work if you only have requester role?

Author: Yes it will.

are found.

## Step 2/3. Create an Access List

Next, we'll create a simple access list that will grant the `access` role to its members.
Login as the administrative user mentioned in the prerequisites. Navigate to the management pane and
click on access lists. Click on "Create an Access List."

![Navigate to create new Access List](../../../img/access-controls/access-lists/create-new-access-list.png)

Here, fill in a title, description, and grant the `access` role.

> ![Fill out Access List Fields](../../../img/access-controls/access-lists/fill-out-access-list-fields.png)

Author: I added this to `reference.mdx`. Let me know what you think.

Under "List Owners" select `editor` as a required role, then add your administrative user under "Add
Eligible List Owners."

![Select an owner](../../../img/access-controls/access-lists/select-owner.png)

Under "Members" select `requester` as a required role, then add your test user to the access list.

![Add a member](../../../img/access-controls/access-lists/add-member.png)

We'll leave the the requirements fields blank and then click "Create Access List" at the bottom
of the page.

## Step 3/3. Login as the test user

Again, login as the test user. When logging in now, you should now see the resources contained within
the cluster, and should be able to interact with them as expected.

> ## Next Steps

Reviewer 2: We should add a link to Access Lists to the "next steps" for access requests guides too.

---
title: Getting Started with Access Lists
description: Learn how to use Access Lists to manage and audit long lived access to Teleport managed resources.
---

This guide will help you:
- Create an access list
- Assign a member to it
- Verify permissions granted through the list membership

## Prerequisites

(!docs/pages/includes/commercial-prereqs-tabs.mdx!)

> - (!docs/pages/includes/tctl.mdx!)

Author: Yes it does.

  enabled: yes
  debug_app: true
```

And restart Teleport. The "dumper" app should show up in the resource list.

![Debug app](../../../img/access-controls/access-lists/debug-app.png)

## Step 2/4. Create a test user

We need to create a simple test user that has only the `requester` role, which has no default access
to anything within a cluster. This user will only be used for the purposes of this guide, so you may use
another user if you so choose. If you would rather use your own user, skip to the next step.

Navigate to the management pane and select "Users." Click on "Create New User" and fill in `test-user` as
the name and select `requester` as the role.

![Create Test User](../../../img/access-controls/access-lists/create-test-user.png)

Click "Save," and then navigate to the provided URL in order to set up the credentials for your test user.
Try logging into the cluster with the test user to verify that no resources show up in the resources page.

## Step 3/4. Create an Access List

Next, we'll create a simple access list that will grant the `access` role to its members.
Login as the administrative user mentioned in the prerequisites. Navigate to the management pane and

> click on access lists. Click on "Create an Access List."

Reviewer 2: ```suggestion
click on access lists. Click on "Create an Access List".
```

Author: It's actually considered more proper to put the period inside the quotes. https://www.hamilton.edu/academics/centers/writing/style/essentials/punctuation-of-quotations#:~:text=The%20final%20period%20or%20comma,the%20period%20follows%20the%20citation.

Reviewer 2: Huh, well, I can't argue with an .edu link 😄

![Navigate to create new Access List](../../../img/access-controls/access-lists/create-new-access-list.png)

Here, fill in a title, description, and grant the `access` role.

![Fill out Access List Fields](../../../img/access-controls/access-lists/fill-out-access-list-fields.png)

> Under "List Owners" select `editor` as a required role, then add your administrative user under "Add

Reviewer: Can we describe what making `editor` a required role actually does? I assume it means anyone with `editor` can manage the access list?

Author: I've described the requirements here and in membership.

Eligible List Owners."

![Select an owner](../../../img/access-controls/access-lists/select-owner.png)

> Under "Members" select `requester` as a required role, then add your test user to the access list.

Author: I've clarified this.

![Add a member](../../../img/access-controls/access-lists/add-member.png)

> We'll leave the the requirements fields blank and then click "Create Access List" at the bottom

Author: This actually doesn't apply any more, so I'll excise this.
